presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -671,6 +735,36 @@ private PlanBuilder window(PlanBuilder subPlan, QuerySpecification node)
         return subPlan;
     }
 
+    private Window rewriteGroupingOperationsInWindowNode(Window window, PlanBuilder subPlan, QuerySpecification node)
```

Comment:
Good suggestion, I've moved all the re-writing parts to `GroupingOperationRewriter`.